src/ast/mod.rs

```diff
@@ -2175,6 +2191,100 @@ impl fmt::Display for Statement {
                 write!(f, "ON {} ", on)?;
                 write!(f, "{}", display_separated(clauses, " "))
             }
+            Statement::CreateSequence {
+                temporary,
+                if_not_exists,
+                name,
+                data_type,
+                sequence_options,
+                owned_by,
+            } => {
+                let mut as_type: String = String::from("");
+                if let Some(dt) = data_type.as_ref() {
+                    as_type = " AS ".to_string() + &dt.to_string();
+                }
+                write!(
+                    f,
+                    "CREATE {temporary}SEQUENCE {if_not_exists}{name}{as_type}",
+                    if_not_exists = if *if_not_exists { "IF NOT EXISTS " } else { "" },
+                    temporary = if *temporary { "TEMPORARY " } else { "" },
+                    name = name,
+                    as_type = as_type
+                )?;
+                for sequence_option in sequence_options {
+                    write!(f, "{}", sequence_option)?;
+                }
+                if let Some(ob) = owned_by.as_ref() {
+                    write!(f, " OWNED BY {}", ob)?;
+                }
+                write!(f, "")
+            }
+        }
+    }
+}
+
+/// Will not add to vec when there are no value since all blocks are optional
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+// #[non_exhaustive]
+pub enum SequenceOptions {
+    ///IncrementBy(increment, by) 'BY' is optional in PostgreSQL
+    IncrementBy(Expr, bool),
+    ///MinValue(minvalue, 'NO MINVALUE')
+    MinValue(Option<Expr>, Option<bool>),
+    ///MaxValue(maxvalue, 'NO MAXVALUE')
+    MaxValue(Option<Expr>, Option<bool>),
+    ///StartWith(start, with) 'WITH' is optional in PostgreSQL
+    StartWith(Expr, bool),
+    Cache(Expr),
+    Cycle(bool),
+}
+
+/// https://www.postgresql.org/docs/current/sql-createsequence.html
+/// Will not add to vec when there are no value since all blocks are optional
+impl fmt::Display for SequenceOptions {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            SequenceOptions::IncrementBy(increment, by) => {
+                write!(
+                    f,
+                    " INCREMENT{by} {increment}",
+                    by = if *by { " BY" } else { "" },
+                    increment = increment
+                )
+            }
+            SequenceOptions::MinValue(minvalue, no) => {
+                if let Some(minvalue) = minvalue.as_ref() {
+                    write!(f, " MINVALUE {minvalue}", minvalue = minvalue)
+                } else if no.is_some() {
+                    write!(f, " NO MINVALUE")
+                } else {
+                    write!(f, "")
+                }
```

Comment:
You can write this type of code more concisely (and have the compiler verify you covered all the cases) like:

```suggestion
                match minvalue.as_ref() {
                    (Some(minvalue), _) =>  write!(f, " MINVALUE {minvalue}", minvalue = minvalue),
                    (None, Some(_)) => write!(f, " NO MINVALUE")
                    (None, _) => write!(f, " NO MINVALUE")
  
```